lib/shadow.c

```diff
@@ -305,8 +305,7 @@ static struct spwd *my_sgetspent (const char *string)
 	if (fields[8][0] == '\0') {
 		spwd.sp_flag = SHADOW_SP_FLAG_UNSET;
 	} else {
-		if (getlong (fields[8], &spwd.sp_flag) == 0) {
-			/* FIXME: add a getulong function */
+		if (getulong (fields[8], &spwd.sp_flag) == 0) {
```

Comment:
Is that the right thing to do?  spwd.sp_flag is a signed int, right?  Would it be better to have a get_int_range(char *in, int *result, int min, mint max);

Reply:
In Fedora the structure is defined in glibc and it's an 'unsigned long int', but I guess that it could change for other distributions. Should I drop this change?

Reply:
Oh, hm.  I guess I was looking at something like man/ja/man3/shadow.3.  I resent the fact that /usr/include/shadow.h has the { on the next line so my "grep -nrH "struct spwd {" /usr/include" didn't catch it :)

Musl also has unsigned (https://git.musl-libc.org/cgit/musl/tree/include/shadow.h).

Let's leave the patch as is, then, thanks.